# Setup

In [1]:
#Import necessary packages

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline
sns.set_style('ticks')

#Read in the data

responses = pd.read_csv('/Users/georgesmith/Desktop/RTP/Survey/ReOpening Survey Items/ReOpening_Survey_Responses_Sept10.csv', header=0)
pd.set_option('display.max_columns', None)
responses.head()

Timestamp  \
0  9/1/20 17:25   
1  9/1/20 17:28   
2  9/1/20 17:28   
3  9/1/20 17:28   
4  9/1/20 17:28   

  Masks or other face coverings required of all individuals (staff and patrons) for admittance.  \
0                               ABSOLUTELY ESSENTIAL                                              
1                                     VERY IMPORTANT                                              
2                               ABSOLUTELY ESSENTIAL                                              
3                               ABSOLUTELY ESSENTIAL                                              
4                               ABSOLUTELY ESSENTIAL                                              

  Proper mask-wearing (both noses and mouthes must be covered) enforced at all times EXCEPT when consuming bar refreshments at your seat.  \
0                               ABSOLUTELY ESSENTIAL                                                                                        
1                                     VERY IMPORTANT                                                                                        
2                               ABSOLUTELY ESSENTIAL                                                                                        
3                               ABSOLUTELY ESSENTIAL                                                                                        
4                               ABSOLUTELY ESSENTIAL                                                                                        

  Reduced seating capacity from 99 to the 33-seat / 11-group floor plan illustrated below.  \
0                                     VERY IMPORTANT                                         
1                                     VERY IMPORTANT                                         
2                               ABSOLUTELY ESSENTIAL                                         
3                              Of AVERAGE IMPORTANCE                                         
4                               ABSOLUTELY ESSENTIAL                                         

  Reduced seating capacity from 99 to the 27-seat / 14-group floor plan illustrated below.  \
0                                     VERY IMPORTANT                                         
1                                     VERY IMPORTANT                                         
2                               ABSOLUTELY ESSENTIAL                                         
3                              Of AVERAGE IMPORTANCE                                         
4                              Of AVERAGE IMPORTANCE                                         

  Increased ventilation by leaving the curtain partition open throughout each performance.   \
0                               NOT IMPORTANT at All                                          
1                                     VERY IMPORTANT                                          
2                               ABSOLUTELY ESSENTIAL                                          
3                                     VERY IMPORTANT                                          
4                               ABSOLUTELY ESSENTIAL                                          

  Standing points in lobby marked 6-feet apart.  \
0                          ABSOLUTELY ESSENTIAL   
1                          ABSOLUTELY ESSENTIAL   
2                          ABSOLUTELY ESSENTIAL   
3                                VERY IMPORTANT   
4                          ABSOLUTELY ESSENTIAL   

  At least 10-feet of social distance between performers (unmasked) and patrons.  \
0                               ABSOLUTELY ESSENTIAL                               
1                                     VERY IMPORTANT                               
2                               ABSOLUTELY ESSENTIAL                               
3                               ABSOLUTELY ESSENTIAL                               
4                    

In [2]:
responses.columns = ['timestamp', 'mandatory_masks', 'proper_mask_wearing', 'reduced_seating_33', 'reduced_seating_27', 'open_curtain', 'standing_points', 'ten_ft_from_cast', 'require_sanitization_on_entry', 'box_office_sanitizer', 'restroom_sanitizer', 'weekly_cast_testing', 'weekly_deep_clean', 'bar_item_cleaning', 'preassigned_entry_times', 'door_tix_scanning', 'mobile_orders_only', 'order_delivery_option', 'cast_temps_preshow', 'patron_temps_preshow', 'attendance_list', 'two_max_restroom', 'restroom_cleaning', 'avg_weekly_outings', 'return_readiness', 'comments']
responses.columns

Index(['timestamp', 'mandatory_masks', 'proper_mask_wearing',
       'reduced_seating_33', 'reduced_seating_27', 'open_curtain',
       'standing_points', 'ten_ft_from_cast', 'require_sanitization_on_entry',
       'box_office_sanitizer', 'restroom_sanitizer', 'weekly_cast_testing',
       'weekly_deep_clean', 'bar_item_cleaning', 'preassigned_entry_times',
       'door_tix_scanning', 'mobile_orders_only', 'order_delivery_option',
       'cast_temps_preshow', 'patron_temps_preshow', 'attendance_list',
       'two_max_restroom', 'restroom_cleaning', 'avg_weekly_outings',
       'return_readiness', 'comments'],
      dtype='object')

In [3]:
responses.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 311 entries, 0 to 310
Data columns (total 26 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   timestamp                      311 non-null    object
 1   mandatory_masks                311 non-null    object
 2   proper_mask_wearing            311 non-null    object
 3   reduced_seating_33             311 non-null    object
 4   reduced_seating_27             311 non-null    object
 5   open_curtain                   311 non-null    object
 6   standing_points                311 non-null    object
 7   ten_ft_from_cast               310 non-null    object
 8   require_sanitization_on_entry  311 non-null    object
 9   box_office_sanitizer           311 non-null    object
 10  restroom_sanitizer             311 non-null    object
 11  weekly_cast_testing            311 non-null    object
 12  weekly_deep_clean              311 non-null    object
 13  bar_i

# Explore responses

In [4]:
responses.describe().T

count unique  \
timestamp                       311    229   
mandatory_masks                 311      5   
proper_mask_wearing             311      5   
reduced_seating_33              311      5   
reduced_seating_27              311      5   
open_curtain                    311      5   
standing_points                 311      5   
ten_ft_from_cast                310      5   
require_sanitization_on_entry   311      5   
box_office_sanitizer            311      5   
restroom_sanitizer              311      5   
weekly_cast_testing             311      5   
weekly_deep_clean               311      5   
bar_item_cleaning               311      5   
preassigned_entry_times         311      5   
door_tix_scanning               311      5   
mobile_orders_only              311      5   
order_delivery_option           311      5   
cast_temps_preshow              311      5   
patron_temps_preshow            311      5   
attendance_list                 311      5   
two_max_restroom                311      5   
restroom_cleaning               311      5   
avg_weekly_outings              311      5   
return_readiness                311      5   
comments                        111    102   

                                                                             top  \
timestamp                                                           9/1/20 17:36   
mandatory_masks                                             ABSOLUTELY ESSENTIAL   
proper_mask_wearing                                         ABSOLUTELY ESSENTIAL   
reduced_seating_33                                                VERY IMPORTANT   
reduced_seating_27                                                VERY IMPORTANT   
open_curtain                                                      VERY IMPORTANT   
standing_points                                                   VERY IMPORTANT   
ten_ft_from_cast                                                  VERY IMPORTANT   
require_sanitization_on_entry                               ABSOLUTELY ESSENTIAL   
box_office_sanitizer                                              VERY IMPORTANT   
restroom_sanitizer                                                VERY IMPORTANT   
weekly_cast_testing                                         ABSOLUTELY ESSENTIAL   
weekly_deep_clean                                           ABSOLUTELY ESSENTIAL   
bar_item_cleaning                                           ABSOLUTELY ESSENTIAL   
preassigned_entry_times                                           VERY IMPORTANT   
door_tix_scanning                                                 VERY IMPORTANT   
mobile_orders_only                                         Of AVERAGE IMPORTANCE   
order_delivery_option                                      Of AVERAGE IMPORTANCE   
cast_temps_preshow                                          ABSOLUTELY ESSENTIAL   
patron_temps_preshow                                              VERY IMPORTANT   
attendance_list                                             ABSOLUTELY ESSENTIAL   
two_max_restroom                                                  VERY IMPORTANT   
restroom_cleaning                                           ABSOLUTELY ESSENTIAL   
avg_weekly_outings                            LESS THAN ONCE PER WEEK on average   
return_readiness                   I am SOMEWHAT READY to attend in-peson events   
comments                       Thank you for your diligence. Excellent questi...   

                              freq  
timestamp                        6  
mandatory_masks                246  
proper_mask_wearing            224  
reduced_seating_33             135  
reduced_seating_27             129  
open_curtain                   102  
standing_points                130  
ten_ft_from_cast               126  
require_sanitization_on_entry  107  
box_office_sanitizer           119  
restroom_sanitizer             125  
weekly_cast_testing            127  
weekly_deep_clean              165  


In summarizing the responses df, we see that the question ten_ft_from_cast has one fewer response than the other questions.
Investigate the cause.

In [5]:
responses[responses.ten_ft_from_cast.isna()]

timestamp       mandatory_masks   proper_mask_wearing  \
134  9/1/20 19:13  ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL   

    reduced_seating_33 reduced_seating_27    open_curtain standing_points  \
134     VERY IMPORTANT     VERY IMPORTANT  VERY IMPORTANT  VERY IMPORTANT   

    ten_ft_from_cast require_sanitization_on_entry box_office_sanitizer  \
134              NaN          ABSOLUTELY ESSENTIAL       VERY IMPORTANT   

    restroom_sanitizer   weekly_cast_testing weekly_deep_clean  \
134     VERY IMPORTANT  ABSOLUTELY ESSENTIAL    VERY IMPORTANT   

        bar_item_cleaning preassigned_entry_times door_tix_scanning  \
134  ABSOLUTELY ESSENTIAL          VERY IMPORTANT    VERY IMPORTANT   

    mobile_orders_only  order_delivery_option cast_temps_preshow  \
134     VERY IMPORTANT  Of AVERAGE IMPORTANCE     VERY IMPORTANT   

    patron_temps_preshow attendance_list      two_max_restroom  \
134       VERY IMPORTANT  VERY IMPORTANT  ABSOLUTELY ESSENTIAL   

        restroom_cleaning                  avg_weekly_outings  \
134  ABSOLUTELY ESSENTIAL  LESS THAN ONCE PER WEEK on average   

                                      return_readiness  \
134  I am DEFINITELY NOT READY to attend any in-per...   

                                              comments  
134  I’d love to support the theater, actors, direc...

ten_ft_from_cast is null for row 134 because we failed to require a response for this question and the respondent did not fill it in. Rather than impute, we'll drop this cell from the analysis.

In [6]:
#Subset "comments" column for future examination
comment_responses = responses.iloc[:, 25]
comment_responses.head()

0                                                  NaN
1                                                  NaN
2    Can't wait to see live theatre again!  Proper ...
3         Thank you for asking great questiond. Bravo!
4                                                  NaN
Name: comments, dtype: object

In [7]:
#Select only the rows for which 'comments' is populated -- keep the current row index to maintain link with the corresponding respondents full record of responses.
comment_responses = comment_responses[comment_responses.notnull()]
comment_responses.head()

2     Can't wait to see live theatre again!  Proper ...
3          Thank you for asking great questiond. Bravo!
6     I could srf incorporating lots of COVID into h...
7     I understnad flexibility will be the key to re...
12    I hope it all works out soon - love coming to ...
Name: comments, dtype: object

In [8]:
#Subset 'avg_weekly_outings' column from the COVID-19 precaution ratings
awo_responses = responses['avg_weekly_outings']
awo_responses.value_counts().reindex(['NEVER', 'LESS THAN ONCE PER WEEK on average', 'ONCE PER WEEK on average', '2 or 3 TIMES PER WEEK on average', '4 OR MORE TIMES PER WEEK on average'])

NEVER                                   50
LESS THAN ONCE PER WEEK on average     102
ONCE PER WEEK on average                79
2 or 3 TIMES PER WEEK on average        57
4 OR MORE TIMES PER WEEK on average     23
Name: avg_weekly_outings, dtype: int64

In [9]:
#Subset 'avg_weekly_outings' column from the COVID-19 precaution ratings
awo_responses = responses['avg_weekly_outings']
awo_responses.value_counts(normalize=True).reindex(['NEVER', 'LESS THAN ONCE PER WEEK on average', 'ONCE PER WEEK on average', '2 or 3 TIMES PER WEEK on average', '4 OR MORE TIMES PER WEEK on average'])

NEVER                                  0.160772
LESS THAN ONCE PER WEEK on average     0.327974
ONCE PER WEEK on average               0.254019
2 or 3 TIMES PER WEEK on average       0.183280
4 OR MORE TIMES PER WEEK on average    0.073955
Name: avg_weekly_outings, dtype: float64

In [10]:
#Subset 'return_readiness' column from the COVID-19 precaution ratings
rr_responses = responses['return_readiness']
rr_responses.value_counts().reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.'], axis='columns')

I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.     34
I am PROBABLY NOT READY to attend in-person events.                                      66
I am SOMEWHAT READY to attend in-peson events                                           104
I am PROBABLY READY to attend in-person events.                                          55
I am DEFINITELY READY to attend in-person events.                                        52
Name: return_readiness, dtype: int64

In [11]:
#Subset 'return_readiness' column from the COVID-19 precaution ratings
rr_responses = responses['return_readiness']
rr_responses.value_counts(normalize=True).reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.'], axis='columns')

I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.    0.109325
I am PROBABLY NOT READY to attend in-person events.                                     0.212219
I am SOMEWHAT READY to attend in-peson events                                           0.334405
I am PROBABLY READY to attend in-person events.                                         0.176849
I am DEFINITELY READY to attend in-person events.                                       0.167203
Name: return_readiness, dtype: float64

In [12]:
#Select remaining columns (less timestamp) for the rating_responses df
rating_responses = responses.iloc[:, 1:25]
rating_responses.describe().T

count unique  \
mandatory_masks                 311      5   
proper_mask_wearing             311      5   
reduced_seating_33              311      5   
reduced_seating_27              311      5   
open_curtain                    311      5   
standing_points                 311      5   
ten_ft_from_cast                310      5   
require_sanitization_on_entry   311      5   
box_office_sanitizer            311      5   
restroom_sanitizer              311      5   
weekly_cast_testing             311      5   
weekly_deep_clean               311      5   
bar_item_cleaning               311      5   
preassigned_entry_times         311      5   
door_tix_scanning               311      5   
mobile_orders_only              311      5   
order_delivery_option           311      5   
cast_temps_preshow              311      5   
patron_temps_preshow            311      5   
attendance_list                 311      5   
two_max_restroom                311      5   
restroom_cleaning               311      5   
avg_weekly_outings              311      5   
return_readiness                311      5   

                                                                         top  \
mandatory_masks                                         ABSOLUTELY ESSENTIAL   
proper_mask_wearing                                     ABSOLUTELY ESSENTIAL   
reduced_seating_33                                            VERY IMPORTANT   
reduced_seating_27                                            VERY IMPORTANT   
open_curtain                                                  VERY IMPORTANT   
standing_points                                               VERY IMPORTANT   
ten_ft_from_cast                                              VERY IMPORTANT   
require_sanitization_on_entry                           ABSOLUTELY ESSENTIAL   
box_office_sanitizer                                          VERY IMPORTANT   
restroom_sanitizer                                            VERY IMPORTANT   
weekly_cast_testing                                     ABSOLUTELY ESSENTIAL   
weekly_deep_clean                                       ABSOLUTELY ESSENTIAL   
bar_item_cleaning                                       ABSOLUTELY ESSENTIAL   
preassigned_entry_times                                       VERY IMPORTANT   
door_tix_scanning                                             VERY IMPORTANT   
mobile_orders_only                                     Of AVERAGE IMPORTANCE   
order_delivery_option                                  Of AVERAGE IMPORTANCE   
cast_temps_preshow                                      ABSOLUTELY ESSENTIAL   
patron_temps_preshow                                          VERY IMPORTANT   
attendance_list                                         ABSOLUTELY ESSENTIAL   
two_max_restroom                                              VERY IMPORTANT   
restroom_cleaning                                       ABSOLUTELY ESSENTIAL   
avg_weekly_outings                        LESS THAN ONCE PER WEEK on average   
return_readiness               I am SOMEWHAT READY to attend in-peson events   

                              freq  
mandatory_masks                246  
proper_mask_wearing            224  
reduced_seating_33             135  
reduced_seating_27             129  
open_curtain                   102  
standing_points                130  
ten_ft_from_cast               126  
require_sanitization_on_entry  107  
box_office_sanitizer           119  
restroom_sanitizer             125  
weekly_cast_testing            127  
weekly_deep_clean              165  
bar_item_cleaning              189  
preassigned_entry_times        141  
door_tix_scanning              114  
mobile_orders_only             128  
order_delivery_option          155  
cast_temps_preshow             117  
patron_temps_preshow           108  
attendance_list                139  
two_max_restroom               122  
restroom_cleaning              201  
avg_weekly_outings         

In [13]:
#Review crosstabs of avg_weekly_outings by return_readiness
pd.crosstab(responses['avg_weekly_outings'], responses.return_readiness, dropna=False, margins=True).reindex(['NEVER', 'LESS THAN ONCE PER WEEK on average', 'ONCE PER WEEK on average', '2 or 3 TIMES PER WEEK on average', '4 OR MORE TIMES PER WEEK on average', 'All' ]).reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.', 'All'], axis='columns')

return_readiness                     I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.  \
avg_weekly_outings                                                                                                          
NEVER                                                                               13                                      
LESS THAN ONCE PER WEEK on average                                                  17                                      
ONCE PER WEEK on average                                                             4                                      
2 or 3 TIMES PER WEEK on average                                                     0                                      
4 OR MORE TIMES PER WEEK on average                                                  0                                      
All                                                                                 34                                      

return_readiness                     I am PROBABLY NOT READY to attend in-person events.  \
avg_weekly_outings                                                                         
NEVER                                                                               18     
LESS THAN ONCE PER WEEK on average                                                  19     
ONCE PER WEEK on average                                                            17     
2 or 3 TIMES PER WEEK on average                                                     7     
4 OR MORE TIMES PER WEEK on average                                                  5     
All                                                                                 66     

return_readiness                     I am SOMEWHAT READY to attend in-peson events  \
avg_weekly_outings                                                                   
NEVER                                                                           14   
LESS THAN ONCE PER WEEK on average                                              37   
ONCE PER WEEK on average                                                        34   
2 or 3 TIMES PER WEEK on average                                                14   
4 OR MORE TIMES PER WEEK on average                                              5   
All                                                                            104   

return_readiness                     I am PROBABLY READY to attend in-person events.  \
avg_weekly_outings                                                                     
NEVER                                                                              4   
LESS THAN ONCE PER WEEK on average                                                16   
ONCE PER WEEK on average                                                          14   
2 or 3 TIMES PER WEEK on average                                                  17   
4 OR MORE TIMES PER WEEK on average                                                4   
All                                                                               55   

return_readiness                     I am DEFINITELY READY to attend in-person events.  \
avg_weekly_outings                                                                       
NEVER                                                                                1   
LESS THAN ONCE PER WEEK on average                                                  13   
ONCE PER WEEK on average                                                            10   
2 or 3 TIMES PER WEEK on average                                                    19   
4 OR MORE TIMES PER WEEK on average                                                  9   
All                                                                                 52   

return_readiness                     All  
avg_weekly_outings                        
NEVER                                 50  
LESS THAN ONCE PER

In [14]:
pd.crosstab(responses['avg_weekly_outings'], responses.return_readiness, dropna=False, margins=True, normalize=True).reindex(['NEVER', 'LESS THAN ONCE PER WEEK on average', 'ONCE PER WEEK on average', '2 or 3 TIMES PER WEEK on average', '4 OR MORE TIMES PER WEEK on average', 'All' ]).reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.', 'All'], axis='columns')

return_readiness                     I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.  \
avg_weekly_outings                                                                                                          
NEVER                                                                         0.041801                                      
LESS THAN ONCE PER WEEK on average                                            0.054662                                      
ONCE PER WEEK on average                                                      0.012862                                      
2 or 3 TIMES PER WEEK on average                                              0.000000                                      
4 OR MORE TIMES PER WEEK on average                                           0.000000                                      
All                                                                           0.109325                                      

return_readiness                     I am PROBABLY NOT READY to attend in-person events.  \
avg_weekly_outings                                                                         
NEVER                                                                         0.057878     
LESS THAN ONCE PER WEEK on average                                            0.061093     
ONCE PER WEEK on average                                                      0.054662     
2 or 3 TIMES PER WEEK on average                                              0.022508     
4 OR MORE TIMES PER WEEK on average                                           0.016077     
All                                                                           0.212219     

return_readiness                     I am SOMEWHAT READY to attend in-peson events  \
avg_weekly_outings                                                                   
NEVER                                                                     0.045016   
LESS THAN ONCE PER WEEK on average                                        0.118971   
ONCE PER WEEK on average                                                  0.109325   
2 or 3 TIMES PER WEEK on average                                          0.045016   
4 OR MORE TIMES PER WEEK on average                                       0.016077   
All                                                                       0.334405   

return_readiness                     I am PROBABLY READY to attend in-person events.  \
avg_weekly_outings                                                                     
NEVER                                                                       0.012862   
LESS THAN ONCE PER WEEK on average                                          0.051447   
ONCE PER WEEK on average                                                    0.045016   
2 or 3 TIMES PER WEEK on average                                            0.054662   
4 OR MORE TIMES PER WEEK on average                                         0.012862   
All                                                                         0.176849   

return_readiness                     I am DEFINITELY READY to attend in-person events.  \
avg_weekly_outings                                                                       
NEVER                                                                         0.003215   
LESS THAN ONCE PER WEEK on average                                            0.041801   
ONCE PER WEEK on average                                                      0.032154   
2 or 3 TIMES PER WEEK on average                                              0.061093   
4 OR MORE TIMES PER WEEK on average                                           0.028939   
All                                                                           0.167203   

return_readiness                          All  
avg_weekly_outings                             
NEVER                                0.160772  
LES

In [15]:
#How are the return_readiness responses distributed across avg_weekly_outing responses?
pd.crosstab(responses['avg_weekly_outings'], responses.return_readiness, dropna=False, margins=True, normalize='columns').reindex(['NEVER', 'LESS THAN ONCE PER WEEK on average', 'ONCE PER WEEK on average', '2 or 3 TIMES PER WEEK on average', '4 OR MORE TIMES PER WEEK on average']).reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.', 'All'], axis='columns')

return_readiness                     I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.  \
avg_weekly_outings                                                                                                          
NEVER                                                                         0.382353                                      
LESS THAN ONCE PER WEEK on average                                            0.500000                                      
ONCE PER WEEK on average                                                      0.117647                                      
2 or 3 TIMES PER WEEK on average                                              0.000000                                      
4 OR MORE TIMES PER WEEK on average                                           0.000000                                      

return_readiness                     I am PROBABLY NOT READY to attend in-person events.  \
avg_weekly_outings                                                                         
NEVER                                                                         0.272727     
LESS THAN ONCE PER WEEK on average                                            0.287879     
ONCE PER WEEK on average                                                      0.257576     
2 or 3 TIMES PER WEEK on average                                              0.106061     
4 OR MORE TIMES PER WEEK on average                                           0.075758     

return_readiness                     I am SOMEWHAT READY to attend in-peson events  \
avg_weekly_outings                                                                   
NEVER                                                                     0.134615   
LESS THAN ONCE PER WEEK on average                                        0.355769   
ONCE PER WEEK on average                                                  0.326923   
2 or 3 TIMES PER WEEK on average                                          0.134615   
4 OR MORE TIMES PER WEEK on average                                       0.048077   

return_readiness                     I am PROBABLY READY to attend in-person events.  \
avg_weekly_outings                                                                     
NEVER                                                                       0.072727   
LESS THAN ONCE PER WEEK on average                                          0.290909   
ONCE PER WEEK on average                                                    0.254545   
2 or 3 TIMES PER WEEK on average                                            0.309091   
4 OR MORE TIMES PER WEEK on average                                         0.072727   

return_readiness                     I am DEFINITELY READY to attend in-person events.  \
avg_weekly_outings                                                                       
NEVER                                                                         0.019231   
LESS THAN ONCE PER WEEK on average                                            0.250000   
ONCE PER WEEK on average                                                      0.192308   
2 or 3 TIMES PER WEEK on average                                              0.365385   
4 OR MORE TIMES PER WEEK on average                                           0.173077   

return_readiness                          All  
avg_weekly_outings                             
NEVER                                0.160772  
LESS THAN ONCE PER WEEK on average   0.327974  
ONCE PER WEEK on average             0.254019  
2 or 3 TIMES PER WEEK on average     0.183280  
4 OR MORE TIMES PER WEEK on average  0.073955

In [16]:
#How are the avg_weekly_outing responses distributed across return_readiness responses?
pd.crosstab(responses['avg_weekly_outings'], responses.return_readiness, dropna=False, margins=True, normalize='index').reindex(['NEVER', 'LESS THAN ONCE PER WEEK on average', 'ONCE PER WEEK on average', '2 or 3 TIMES PER WEEK on average', '4 OR MORE TIMES PER WEEK on average', 'All' ]).reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.'], axis='columns')

return_readiness                     I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.  \
avg_weekly_outings                                                                                                          
NEVER                                                                         0.260000                                      
LESS THAN ONCE PER WEEK on average                                            0.166667                                      
ONCE PER WEEK on average                                                      0.050633                                      
2 or 3 TIMES PER WEEK on average                                              0.000000                                      
4 OR MORE TIMES PER WEEK on average                                           0.000000                                      
All                                                                           0.109325                                      

return_readiness                     I am PROBABLY NOT READY to attend in-person events.  \
avg_weekly_outings                                                                         
NEVER                                                                         0.360000     
LESS THAN ONCE PER WEEK on average                                            0.186275     
ONCE PER WEEK on average                                                      0.215190     
2 or 3 TIMES PER WEEK on average                                              0.122807     
4 OR MORE TIMES PER WEEK on average                                           0.217391     
All                                                                           0.212219     

return_readiness                     I am SOMEWHAT READY to attend in-peson events  \
avg_weekly_outings                                                                   
NEVER                                                                     0.280000   
LESS THAN ONCE PER WEEK on average                                        0.362745   
ONCE PER WEEK on average                                                  0.430380   
2 or 3 TIMES PER WEEK on average                                          0.245614   
4 OR MORE TIMES PER WEEK on average                                       0.217391   
All                                                                       0.334405   

return_readiness                     I am PROBABLY READY to attend in-person events.  \
avg_weekly_outings                                                                     
NEVER                                                                       0.080000   
LESS THAN ONCE PER WEEK on average                                          0.156863   
ONCE PER WEEK on average                                                    0.177215   
2 or 3 TIMES PER WEEK on average                                            0.298246   
4 OR MORE TIMES PER WEEK on average                                         0.173913   
All                                                                         0.176849   

return_readiness                     I am DEFINITELY READY to attend in-person events.  
avg_weekly_outings                                                                      
NEVER                                                                         0.020000  
LESS THAN ONCE PER WEEK on average                                            0.127451  
ONCE PER WEEK on average                                                      0.126582  
2 or 3 TIMES PER WEEK on average                                              0.333333  
4 OR MORE TIMES PER WEEK on average                                           0.391304  
All                                                                           0.167203

In [17]:
#Review the crosstabs for 'return_readiness' against mandatory_masks
pd.crosstab(rr_responses, rating_responses.mandatory_masks, dropna=False, margins=True).reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.', 'All']).reindex(['NOT IMPORTANT at All', 'Of LITTLE IMPORTANCE', 'Of AVERAGE IMPORTANCE', 'VERY IMPORTANT', 'ABSOLUTELY ESSENTIAL', 'All'], axis='columns')

mandatory_masks                                     NOT IMPORTANT at All  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...                     0   
I am PROBABLY NOT READY to attend in-person eve...                     0   
I am SOMEWHAT READY to attend in-peson events                          0   
I am PROBABLY READY to attend in-person events.                        0   
I am DEFINITELY READY to attend in-person events.                      3   
All                                                                    3   

mandatory_masks                                     Of LITTLE IMPORTANCE  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...                     0   
I am PROBABLY NOT READY to attend in-person eve...                     0   
I am SOMEWHAT READY to attend in-peson events                          1   
I am PROBABLY READY to attend in-person events.                        1   
I am DEFINITELY READY to attend in-person events.                      3   
All                                                                    5   

mandatory_masks                                     Of AVERAGE IMPORTANCE  \
return_readiness                                                            
I am DEFINITELY NOT READY to attend any in-pers...                      0   
I am PROBABLY NOT READY to attend in-person eve...                      0   
I am SOMEWHAT READY to attend in-peson events                           1   
I am PROBABLY READY to attend in-person events.                         1   
I am DEFINITELY READY to attend in-person events.                       7   
All                                                                     9   

mandatory_masks                                     VERY IMPORTANT  \
return_readiness                                                     
I am DEFINITELY NOT READY to attend any in-pers...               1   
I am PROBABLY NOT READY to attend in-person eve...               8   
I am SOMEWHAT READY to attend in-peson events                   11   
I am PROBABLY READY to attend in-person events.                 14   
I am DEFINITELY READY to attend in-person events.               14   
All                                                             48   

mandatory_masks                                     ABSOLUTELY ESSENTIAL  All  
return_readiness                                                               
I am DEFINITELY NOT READY to attend any in-pers...                    33   34  
I am PROBABLY NOT READY to attend in-person eve...                    58   66  
I am SOMEWHAT READY to attend in-peson events                         91  104  
I am PROBABLY READY to attend in-person events.                       39   55  
I am DEFINITELY READY to attend in-person events.                     25   52  
All                                                                  246  311

In [18]:
#Review the normalized crosstabs of return_readiness by mandatory_masks importance
pd.crosstab(rr_responses, rating_responses.mandatory_masks, dropna=False, margins=True, normalize=True).reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.', 'All']).reindex(['NOT IMPORTANT at All', 'Of LITTLE IMPORTANCE', 'Of AVERAGE IMPORTANCE', 'VERY IMPORTANT', 'ABSOLUTELY ESSENTIAL', 'All'], axis='columns')

mandatory_masks                                     NOT IMPORTANT at All  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...              0.000000   
I am PROBABLY NOT READY to attend in-person eve...              0.000000   
I am SOMEWHAT READY to attend in-peson events                   0.000000   
I am PROBABLY READY to attend in-person events.                 0.000000   
I am DEFINITELY READY to attend in-person events.               0.009646   
All                                                             0.009646   

mandatory_masks                                     Of LITTLE IMPORTANCE  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...              0.000000   
I am PROBABLY NOT READY to attend in-person eve...              0.000000   
I am SOMEWHAT READY to attend in-peson events                   0.003215   
I am PROBABLY READY to attend in-person events.                 0.003215   
I am DEFINITELY READY to attend in-person events.               0.009646   
All                                                             0.016077   

mandatory_masks                                     Of AVERAGE IMPORTANCE  \
return_readiness                                                            
I am DEFINITELY NOT READY to attend any in-pers...               0.000000   
I am PROBABLY NOT READY to attend in-person eve...               0.000000   
I am SOMEWHAT READY to attend in-peson events                    0.003215   
I am PROBABLY READY to attend in-person events.                  0.003215   
I am DEFINITELY READY to attend in-person events.                0.022508   
All                                                              0.028939   

mandatory_masks                                     VERY IMPORTANT  \
return_readiness                                                     
I am DEFINITELY NOT READY to attend any in-pers...        0.003215   
I am PROBABLY NOT READY to attend in-person eve...        0.025723   
I am SOMEWHAT READY to attend in-peson events             0.035370   
I am PROBABLY READY to attend in-person events.           0.045016   
I am DEFINITELY READY to attend in-person events.         0.045016   
All                                                       0.154341   

mandatory_masks                                     ABSOLUTELY ESSENTIAL  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...              0.106109   
I am PROBABLY NOT READY to attend in-person eve...              0.186495   
I am SOMEWHAT READY to attend in-peson events                   0.292605   
I am PROBABLY READY to attend in-person events.                 0.125402   
I am DEFINITELY READY to attend in-person events.               0.080386   
All                                                             0.790997   

mandatory_masks                                          All  
return_readiness                                              
I am DEFINITELY NOT READY to attend any in-pers...  0.109325  
I am PROBABLY NOT READY to attend in-person eve...  0.212219  
I am SOMEWHAT READY to attend in-peson events       0.334405  
I am PROBABLY READY to attend in-person events.     0.176849  
I am DEFINITELY READY to attend in-person events.   0.167203  
All                                                 1.000000

In [19]:
pd.crosstab(rr_responses, rating_responses.mandatory_masks, dropna=False, margins=True, normalize='index').reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.', 'All']).reindex(['NOT IMPORTANT at All', 'Of LITTLE IMPORTANCE', 'Of AVERAGE IMPORTANCE', 'VERY IMPORTANT', 'ABSOLUTELY ESSENTIAL'], axis='columns')

mandatory_masks                                     NOT IMPORTANT at All  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...              0.000000   
I am PROBABLY NOT READY to attend in-person eve...              0.000000   
I am SOMEWHAT READY to attend in-peson events                   0.000000   
I am PROBABLY READY to attend in-person events.                 0.000000   
I am DEFINITELY READY to attend in-person events.               0.057692   
All                                                             0.009646   

mandatory_masks                                     Of LITTLE IMPORTANCE  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...              0.000000   
I am PROBABLY NOT READY to attend in-person eve...              0.000000   
I am SOMEWHAT READY to attend in-peson events                   0.009615   
I am PROBABLY READY to attend in-person events.                 0.018182   
I am DEFINITELY READY to attend in-person events.               0.057692   
All                                                             0.016077   

mandatory_masks                                     Of AVERAGE IMPORTANCE  \
return_readiness                                                            
I am DEFINITELY NOT READY to attend any in-pers...               0.000000   
I am PROBABLY NOT READY to attend in-person eve...               0.000000   
I am SOMEWHAT READY to attend in-peson events                    0.009615   
I am PROBABLY READY to attend in-person events.                  0.018182   
I am DEFINITELY READY to attend in-person events.                0.134615   
All                                                              0.028939   

mandatory_masks                                     VERY IMPORTANT  \
return_readiness                                                     
I am DEFINITELY NOT READY to attend any in-pers...        0.029412   
I am PROBABLY NOT READY to attend in-person eve...        0.121212   
I am SOMEWHAT READY to attend in-peson events             0.105769   
I am PROBABLY READY to attend in-person events.           0.254545   
I am DEFINITELY READY to attend in-person events.         0.269231   
All                                                       0.154341   

mandatory_masks                                     ABSOLUTELY ESSENTIAL  
return_readiness                                                          
I am DEFINITELY NOT READY to attend any in-pers...              0.970588  
I am PROBABLY NOT READY to attend in-person eve...              0.878788  
I am SOMEWHAT READY to attend in-peson events                   0.875000  
I am PROBABLY READY to attend in-person events.                 0.709091  
I am DEFINITELY READY to attend in-person events.               0.480769  
All                                                             0.790997

In [20]:
pd.crosstab(rr_responses, rating_responses.mandatory_masks, dropna=False, margins=True, normalize='columns').reindex(['I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.', 'I am PROBABLY NOT READY to attend in-person events.', 'I am SOMEWHAT READY to attend in-peson events', 'I am PROBABLY READY to attend in-person events.', 'I am DEFINITELY READY to attend in-person events.']).reindex(['NOT IMPORTANT at All', 'Of LITTLE IMPORTANCE', 'Of AVERAGE IMPORTANCE', 'VERY IMPORTANT', 'ABSOLUTELY ESSENTIAL', 'All'], axis='columns')

mandatory_masks                                     NOT IMPORTANT at All  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...                   0.0   
I am PROBABLY NOT READY to attend in-person eve...                   0.0   
I am SOMEWHAT READY to attend in-peson events                        0.0   
I am PROBABLY READY to attend in-person events.                      0.0   
I am DEFINITELY READY to attend in-person events.                    1.0   

mandatory_masks                                     Of LITTLE IMPORTANCE  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...                   0.0   
I am PROBABLY NOT READY to attend in-person eve...                   0.0   
I am SOMEWHAT READY to attend in-peson events                        0.2   
I am PROBABLY READY to attend in-person events.                      0.2   
I am DEFINITELY READY to attend in-person events.                    0.6   

mandatory_masks                                     Of AVERAGE IMPORTANCE  \
return_readiness                                                            
I am DEFINITELY NOT READY to attend any in-pers...               0.000000   
I am PROBABLY NOT READY to attend in-person eve...               0.000000   
I am SOMEWHAT READY to attend in-peson events                    0.111111   
I am PROBABLY READY to attend in-person events.                  0.111111   
I am DEFINITELY READY to attend in-person events.                0.777778   

mandatory_masks                                     VERY IMPORTANT  \
return_readiness                                                     
I am DEFINITELY NOT READY to attend any in-pers...        0.020833   
I am PROBABLY NOT READY to attend in-person eve...        0.166667   
I am SOMEWHAT READY to attend in-peson events             0.229167   
I am PROBABLY READY to attend in-person events.           0.291667   
I am DEFINITELY READY to attend in-person events.         0.291667   

mandatory_masks                                     ABSOLUTELY ESSENTIAL  \
return_readiness                                                           
I am DEFINITELY NOT READY to attend any in-pers...              0.134146   
I am PROBABLY NOT READY to attend in-person eve...              0.235772   
I am SOMEWHAT READY to attend in-peson events                   0.369919   
I am PROBABLY READY to attend in-person events.                 0.158537   
I am DEFINITELY READY to attend in-person events.               0.101626   

mandatory_masks                                          All  
return_readiness                                              
I am DEFINITELY NOT READY to attend any in-pers...  0.109325  
I am PROBABLY NOT READY to attend in-person eve...  0.212219  
I am SOMEWHAT READY to attend in-peson events       0.334405  
I am PROBABLY READY to attend in-person events.     0.176849  
I am DEFINITELY READY to attend in-person events.   0.167203

**Convert responses to numeric**

In [21]:
#Remove timestamp and comments columns
responses_num = responses.iloc[:,1:25]
responses_num.head()

mandatory_masks   proper_mask_wearing     reduced_seating_33  \
0  ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL         VERY IMPORTANT   
1        VERY IMPORTANT        VERY IMPORTANT         VERY IMPORTANT   
2  ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL   ABSOLUTELY ESSENTIAL   
3  ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL  Of AVERAGE IMPORTANCE   
4  ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL   ABSOLUTELY ESSENTIAL   

      reduced_seating_27          open_curtain       standing_points  \
0         VERY IMPORTANT  NOT IMPORTANT at All  ABSOLUTELY ESSENTIAL   
1         VERY IMPORTANT        VERY IMPORTANT  ABSOLUTELY ESSENTIAL   
2   ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL   
3  Of AVERAGE IMPORTANCE        VERY IMPORTANT        VERY IMPORTANT   
4  Of AVERAGE IMPORTANCE  ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL   

       ten_ft_from_cast require_sanitization_on_entry   box_office_sanitizer  \
0  ABSOLUTELY ESSENTIAL         Of AVERAGE IMPORTANCE   NOT IMPORTANT at All   
1        VERY IMPORTANT          ABSOLUTELY ESSENTIAL         VERY IMPORTANT   
2  ABSOLUTELY ESSENTIAL          ABSOLUTELY ESSENTIAL   ABSOLUTELY ESSENTIAL   
3  ABSOLUTELY ESSENTIAL         Of AVERAGE IMPORTANCE  Of AVERAGE IMPORTANCE   
4  ABSOLUTELY ESSENTIAL          ABSOLUTELY ESSENTIAL   ABSOLUTELY ESSENTIAL   

      restroom_sanitizer   weekly_cast_testing      weekly_deep_clean  \
0         VERY IMPORTANT  NOT IMPORTANT at All   Of LITTLE IMPORTANCE   
1         VERY IMPORTANT        VERY IMPORTANT   ABSOLUTELY ESSENTIAL   
2   ABSOLUTELY ESSENTIAL        VERY IMPORTANT   ABSOLUTELY ESSENTIAL   
3  Of AVERAGE IMPORTANCE        VERY IMPORTANT  Of AVERAGE IMPORTANCE   
4   ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL         VERY IMPORTANT   

       bar_item_cleaning preassigned_entry_times      door_tix_scanning  \
0  Of AVERAGE IMPORTANCE    ABSOLUTELY ESSENTIAL   NOT IMPORTANT at All   
1   ABSOLUTELY ESSENTIAL          VERY IMPORTANT  Of AVERAGE IMPORTANCE   
2         VERY IMPORTANT    ABSOLUTELY ESSENTIAL   ABSOLUTELY ESSENTIAL   
3   ABSOLUTELY ESSENTIAL    ABSOLUTELY ESSENTIAL  Of AVERAGE IMPORTANCE   
4   ABSOLUTELY ESSENTIAL    ABSOLUTELY ESSENTIAL   ABSOLUTELY ESSENTIAL   

      mobile_orders_only  order_delivery_option     cast_temps_preshow  \
0  Of AVERAGE IMPORTANCE  Of AVERAGE IMPORTANCE  Of AVERAGE IMPORTANCE   
1  Of AVERAGE IMPORTANCE  Of AVERAGE IMPORTANCE  Of AVERAGE IMPORTANCE   
2  Of AVERAGE IMPORTANCE  Of AVERAGE IMPORTANCE   ABSOLUTELY ESSENTIAL   
3   ABSOLUTELY ESSENTIAL         VERY IMPORTANT   ABSOLUTELY ESSENTIAL   
4  Of AVERAGE IMPORTANCE  Of AVERAGE IMPORTANCE   ABSOLUTELY ESSENTIAL   

    patron_temps_preshow       attendance_list       two_max_restroom  \
0  Of AVERAGE IMPORTANCE  Of LITTLE IMPORTANCE   ABSOLUTELY ESSENTIAL   
1  Of AVERAGE IMPORTANCE        VERY IMPORTANT         VERY IMPORTANT   
2   ABSOLUTELY ESSENTIAL        VERY IMPORTANT   ABSOLUTELY ESSENTIAL   
3   ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL  Of AVERAGE IMPORTANCE   
4   ABSOLUTELY ESSENTIAL  ABSOLUTELY ESSENTIAL   ABSOLUTELY ESSENTIAL   

      restroom_cleaning                  avg_weekly_outings  \
0  ABSOLUTELY ESSENTIAL  LESS THAN ONCE PER WEEK on average   
1  ABSOLUTELY ESSENTIAL    2 or 3 TIMES PER WEEK on average   
2  ABSOLUTELY ESSENTIAL  LESS THAN ONCE PER WEEK on average   
3  ABSOLUTELY ESSENTIAL  LESS THAN ONCE PER WEEK on average   
4  ABSOLUTELY ESSENTIAL            ONCE PER WEEK on average   

                                    return_readiness  
0      I am SOMEWHAT READY to attend in-peson events  
1      I am SOMEWHAT READY to attend in-peson events  
2  I am DEFINITELY READY to attend in-person events.  
3      I am SOMEWHAT READY to attend in-peson events  
4  I am DEFINITELY READY to attend in-person events.

In [22]:
rosurvey_dict1 = {
    'NOT IMPORTANT at All' : 1,
    'Of LITTLE IMPORTANCE' : 2,
    'Of AVERAGE IMPORTANCE' : 3,
    'VERY IMPORTANT' : 4,
    'ABSOLUTELY ESSENTIAL' : 5
}

In [23]:
rosurvey_dict2 = {
    'NEVER' : 1,
    'LESS THAN ONCE PER WEEK on average' : 2,
    'ONCE PER WEEK on average' : 3,
    '2 or 3 TIMES PER WEEK on average' : 4,
    '4 OR MORE TIMES PER WEEK on average' : 5
}

In [24]:
rosurvey_dict3 = {
    'I am DEFINITELY NOT READY to attend any in-person events for the foreseeable future.' : 1,
    'I am PROBABLY NOT READY to attend in-person events.' : 2,
    'I am SOMEWHAT READY to attend in-peson events' : 3,
    'I am PROBABLY READY to attend in-person events.' : 4,
    'I am DEFINITELY READY to attend in-person events.' : 5
}

In [25]:
#Map numerics to responses for rating columns for analysis

responses_num = responses_num.replace(rosurvey_dict1)
responses_num.head()

mandatory_masks  proper_mask_wearing  reduced_seating_33  \
0                5                    5                   4   
1                4                    4                   4   
2                5                    5                   5   
3                5                    5                   3   
4                5                    5                   5   

   reduced_seating_27  open_curtain  standing_points  ten_ft_from_cast  \
0                   4             1                5               5.0   
1                   4             4                5               4.0   
2                   5             5                5               5.0   
3                   3             4                4               5.0   
4                   3             5                5               5.0   

   require_sanitization_on_entry  box_office_sanitizer  restroom_sanitizer  \
0                              3                     1                   4   
1                              5                     4                   4   
2                              5                     5                   5   
3                              3                     3                   3   
4                              5                     5                   5   

   weekly_cast_testing  weekly_deep_clean  bar_item_cleaning  \
0                    1                  2                  3   
1                    4                  5                  5   
2                    4                  5                  4   
3                    4                  3                  5   
4                    5                  4                  5   

   preassigned_entry_times  door_tix_scanning  mobile_orders_only  \
0                        5                  1                   3   
1                        4                  3                   3   
2                        5                  5                   3   
3                        5                  3                   5   
4                        5                  5                   3   

   order_delivery_option  cast_temps_preshow  patron_temps_preshow  \
0                      3                   3                     3   
1                      3                   3                     3   
2                      3                   5                     5   
3                      4                   5                     5   
4                      3                   5                     5   

   attendance_list  two_max_restroom  restroom_cleaning  \
0                2                 5                  5   
1                4                 4                  5   
2                4                 5                  5   
3                5                 3                  5   
4                5                 5                  5   

                   avg_weekly_outings  \
0  LESS THAN ONCE PER WEEK on average   
1    2 or 3 TIMES PER WEEK on average   
2  LESS THAN ONCE PER WEEK on average   
3  LESS THAN ONCE PER WEEK on average   
4            ONCE PER WEEK on average   

                                    return_readiness  
0      I am SOMEWHAT READY to attend in-peson events  
1      I am SOMEWHAT READY to attend in-peson events  
2  I am DEFINITELY READY to attend in-person events.  
3      I am SOMEWHAT READY to attend in-peson events  
4  I am DEFINITELY READY to attend in-person events.

In [26]:
#Map numerics to responses in 'avg_weekly_outings' column for analysis

responses_num.avg_weekly_outings = responses_num.avg_weekly_outings.replace(rosurvey_dict2)
responses_num.avg_weekly_outings.head()

0    2
1    4
2    2
3    2
4    3
Name: avg_weekly_outings, dtype: int64

In [27]:
#Map numerics to responses in 'return_readiness' column for analysis

responses_num.return_readiness = responses_num.return_readiness.replace(rosurvey_dict3)
responses_num.return_readiness.head()

0    3
1    3
2    5
3    3
4    5
Name: return_readiness, dtype: int64

In [28]:
responses_num.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 311 entries, 0 to 310
Data columns (total 24 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   mandatory_masks                311 non-null    int64  
 1   proper_mask_wearing            311 non-null    int64  
 2   reduced_seating_33             311 non-null    int64  
 3   reduced_seating_27             311 non-null    int64  
 4   open_curtain                   311 non-null    int64  
 5   standing_points                311 non-null    int64  
 6   ten_ft_from_cast               310 non-null    float64
 7   require_sanitization_on_entry  311 non-null    int64  
 8   box_office_sanitizer           311 non-null    int64  
 9   restroom_sanitizer             311 non-null    int64  
 10  weekly_cast_testing            311 non-null    int64  
 11  weekly_deep_clean              311 non-null    int64  
 12  bar_item_cleaning              311 non-null    int

In [29]:
responses_num.describe().T

count      mean       std  min  25%  50%  75%  \
mandatory_masks                311.0  4.700965  0.693727  1.0  5.0  5.0  5.0   
proper_mask_wearing            311.0  4.601286  0.771591  1.0  4.0  5.0  5.0   
reduced_seating_33             311.0  4.112540  0.935089  1.0  4.0  4.0  5.0   
reduced_seating_27             311.0  3.800643  1.034314  1.0  3.0  4.0  5.0   
open_curtain                   311.0  3.504823  1.158061  1.0  3.0  4.0  4.0   
standing_points                311.0  3.887460  0.945381  1.0  3.0  4.0  5.0   
ten_ft_from_cast               310.0  4.045161  0.974377  1.0  4.0  4.0  5.0   
require_sanitization_on_entry  311.0  3.877814  1.070660  1.0  3.0  4.0  5.0   
box_office_sanitizer           311.0  3.864952  0.944130  1.0  3.0  4.0  5.0   
restroom_sanitizer             311.0  4.032154  0.918762  1.0  4.0  4.0  5.0   
weekly_cast_testing            311.0  4.073955  0.985866  1.0  4.0  4.0  5.0   
weekly_deep_clean              311.0  4.331190  0.855509  1.0  4.0  5.0  5.0   
bar_item_cleaning              311.0  4.475884  0.769518  1.0  4.0  5.0  5.0   
preassigned_entry_times        311.0  3.726688  0.939747  1.0  3.0  4.0  4.0   
door_tix_scanning              311.0  3.469453  1.034033  1.0  3.0  4.0  4.0   
mobile_orders_only             311.0  3.019293  1.031573  1.0  2.0  3.0  4.0   
order_delivery_option          311.0  2.932476  0.956442  1.0  2.0  3.0  3.0   
cast_temps_preshow             311.0  4.022508  0.981839  1.0  3.0  4.0  5.0   
patron_temps_preshow           311.0  3.765273  1.098188  1.0  3.0  4.0  5.0   
attendance_list                311.0  4.073955  1.049269  1.0  3.0  4.0  5.0   
two_max_restroom               311.0  3.980707  0.956951  1.0  3.0  4.0  5.0   
restroom_cleaning              311.0  4.504823  0.790555  1.0  4.0  5.0  5.0   
avg_weekly_outings             311.0  2.681672  1.163262  1.0  2.0  3.0  4.0   
return_readiness               311.0  3.080386  1.222095  1.0  2.0  3.0  4.0   

                               max  
mandatory_masks                5.0  
proper_mask_wearing            5.0  
reduced_seating_33             5.0  
reduced_seating_27             5.0  
open_curtain                   5.0  
standing_points                5.0  
ten_ft_from_cast               5.0  
require_sanitization_on_entry  5.0  
box_office_sanitizer           5.0  
restroom_sanitizer             5.0  
weekly_cast_testing            5.0  
weekly_deep_clean              5.0  
bar_item_cleaning              5.0  
preassigned_entry_times        5.0  
door_tix_scanning              5.0  
mobile_orders_only             5.0  
order_delivery_option          5.0  
cast_temps_preshow             5.0  
patron_temps_preshow           5.0  
attendance_list                5.0  
two_max_restroom               5.0  
restroom_cleaning              5.0  
avg_weekly_outings             5.0  
return_readiness               5.0

In [30]:
#Export as csv
responses_num.to_csv('reopening_survey_responses_Sept10.csv')

In [ ]:
# rating_cats = rating_responses.columns

# for cat in rating_cats:
#     cts = pd.crosstab(awo_responses, rating_responses[cat], dropna=False, normalize=True)
#     print(cts)

In [ ]:
# #Plot rating column counts by awo_response

# ##FIX THE FUNCTION BELOW ACCORDINGLY##

# def plot_counts_by_age_group(groupby_count_obj, age_group, ax=None):
    
#     """Takes a count-aggregated groupby object, an age group, and an 
#     (optional) AxesSubplot, and draws a barplot for that group."""
    
#     sort_order = groupby_count_obj.loc[age_group].sort_index().index

#     sns.barplot(y = groupby_count_obj.loc[age_group].index, 
#                 x = groupby_count_obj.loc[age_group].values, 
#                 order = sort_order, 
#                 palette = 'rocket', edgecolor = 'black', 
#                 ax = ax
#                 ).set_title("Age {}".format(age_group))

In [ ]:
# names = []
# not_important_count = []
# little_importance_count = []
# avg_importance_count = []
# very_important_count = []
# absolutely_essential_count = []
# cols_to_use = rating_responses.columns

# for col in cols_to_use:
#     resp_counts = rating_responses[col].value_counts()
#     names.append(col)
#     not_important_count.append(resp_counts['NOT IMPORTANT at All'])
#     little_importance_count.append(resp_counts['Of LITTLE IMPORTANCE'])
#     avg_importance_count.append(resp_counts['Of AVERAGE IMPORTANCE'])
#     very_important_count.append(resp_counts['VERY IMPORTANT'])
#     absolutely_essential_count.append(resp_counts['ABSOLUTELY ESSENTIAL'])

# trace1 = go.Bar(
#     y=names,
#     x=not_important_count,
#     orientation = 'h',
#     name = 'NOT IMPORTANT at All'
# )
# trace2 = go.Bar(
#     y=names,
#     x=little_importance_count,
#     orientation = 'h',
#     name = 'Of LITTLE IMPORTANCE'
# )
# trace3 = go.Bar(
#     y=names,
#     x=avg_importance_count,
#     orientation = 'h',
#     name = 'Of AVERAGE IMPORTANCE'
# )
# trace4 = go.Bar(
#     y=names,
#     x=very_important_count,
#     orientation = 'h',
#     name = 'VERY IMPORTANT'
# )
# trace5 = go.Bar(
#     y=names,
#     x=absolutely_essential_count,
#     orientation = 'h',
#     name = 'ABSOLUTELY ESSENTIAL'
# )

# layout = go.Layout(
#     title="How Important Are These COVID-19 Precautions to RTP's Audience",
#     barmode='stack',
#     width = 800,
#     height = 2250
#     #yaxis=dict(tickangle=-45),
# )

# data = [trace1, trace2, trace3, trace4, trace5]
# fig = go.Figure(data=data, layout=layout)
# py.iplot(fig, filename="Precaution Importance")